### PR TITLE
554 network stopping

### PR DESCRIPTION
**Pull request by dchertkov**

### CODE REVIEW

File: packages/daylight/start.go [hunk 1]
```diff
@@ -235,6 +243,10 @@ func Start() {
 		if err != nil {
 			os.Exit(1)
 		}
+		//go func() {
```
Comment (potehinre) [inline packages/daylight/start.go]: Commented code prohibited

File: cmd/stopNetwork.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+		}
+
+		req := &tcpserver.StopNetworkRequest{
+			Data: stopNetworkCert,
+		}
+
+		errCount := 0
+		for _, addr := range addrsForStopping {
+			if err := sendStopNetworkCert(addr, req); err != nil {
+				log.WithError(err).Errorf("Sending request to %s", addr)
```
Comment (potehinre) [inline cmd/stopNetwork.go]: better use WithFields, than formatted logging

File: cmd/stopNetwork.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+				errCount++
+				continue
+			}
+
+			log.Infof("Sending request to %s", addr)
+		}
+
+		log.Infof("Successful: %d, failed: %d", len(addrsForStopping)-errCount, errCount)
```
Comment (potehinre) [inline cmd/stopNetwork.go]: the same

File: cmd/stopNetwork.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+		errCount := 0
+		for _, addr := range addrsForStopping {
+			if err := sendStopNetworkCert(addr, req); err != nil {
+				log.WithError(err).Errorf("Sending request to %s", addr)
+				errCount++
+				continue
+			}
+
+			log.Infof("Sending request to %s", addr)
```
Comment (potehinre) [inline cmd/stopNetwork.go]: the same

File: packages/tcpserver/type3.go [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+		return err
+	}
+
+	return nil
+}
+
+func processStopNetwork(b []byte) ([]byte, error) {
+	cert, err := utils.ParseCert(b)
+	if err != nil {
+		log.WithError(err).Error("parsing cert")
```
Comment (potehinre) [inline packages/tcpserver/type3.go]: Here and below logging: error types? ParseError, InvalidObjectError

File: packages/tcpserver/type3.go [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+			TxHeader: consts.TxHeader{
+				Type:  consts.TxTypeStopNetwork,
+				Time:  uint32(time.Now().Unix()),
+				KeyID: conf.Config.KeyID,
+			},
+			StopNetworkCert: b,
+		},
+	)
+	if err != nil {
+		log.WithError(err).Error("binary marshaling")
```
Comment (potehinre) [inline packages/tcpserver/type3.go]: MarshallingError

File: packages/tcpserver/type3.go [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+		return nil, err
+	}
+
+	hash, err := crypto.Hash(data)
+	if err != nil {
+		log.WithError(err).Error("hashing data")
```
Comment (potehinre) [inline packages/tcpserver/type3.go]: CryptoError

File: packages/tcpserver/type3.go [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+	tx := &model.Transaction{
+		Hash:     hash,
+		Data:     data,
+		Type:     consts.TxTypeStopNetwork,
+		KeyID:    conf.Config.KeyID,
+		HighRate: model.TransactionRateStopNetwork,
+	}
+	if err = tx.Create(); err != nil {
+		log.WithError(err).Error("inserting tx to database")
```
Comment (potehinre) [inline packages/tcpserver/type3.go]: DBError

**Review by potehinre — CHANGES_REQUESTED** [review]

File: cmd/stopNetwork.go [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+
+	"github.com/GenesisKernel/go-genesis/packages/converter"
+	"github.com/GenesisKernel/go-genesis/packages/tcpserver"
+	"github.com/GenesisKernel/go-genesis/packages/utils"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+const reqType = 3
```
Comment (ghost) [inline cmd/stopNetwork.go]: what does it mean? Add a comment or give more semantic name.

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by potehinre — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by potehinre — APPROVED** [review]

**Pull request merged**